server/utils/isValidFindSlugParams.js

```diff
@@ -15,7 +16,9 @@ const isValidFindSlugParams = (params) => {
 		throw new ValidationError('A slug path variable is required.');
 	}
 
-	const model = models[modelName];
+	if (!model.contentType.options.draftAndPublish && publicationState) {
```

Comment:
I would use the lodash check here same as in the types file, otherwise it requires additional check for options as it is not gaurunteed to exist. Lodash would also need to be imported here ofc.

```suggestion
	if (_.get(contentType, ['options', 'draftAndPublish'], false) && publicationState) {
```